--- monit.h.original	2024-08-29 14:45:56.768393000 +0200
+++ monit.h	2024-08-29 14:48:47.816825000 +0200
@@ -22,6 +22,32 @@
  * for all of the code used other than OpenSSL.
  */
 
+// TO CREATE THE DIFF FILE: diff -Naru monit.h.original monit.h > monit.h.patch 
+// CYGWIN ADDED LINES START 
+#ifdef CYGWIN
+#ifndef ICMP_ECHO
+#define ICMP_ECHO 8
+#endif
+#ifndef ICMP6_ECHO
+#define ICMP6_ECHO 8
+#endif
+#ifndef ICMP_ECHOREPLY
+#define ICMP_ECHOREPLY 20
+#endif
+#ifndef ICMP6_ECHO_REPLY
+#define ICMP6_ECHO_REPLY 20
+#endif
+#ifndef ICMP6_ECHO_REQUEST
+#define ICMP6_ECHO_REQUEST 128
+#endif
+#ifndef ICMP6_FILTER
+#define ICMP6_FILTER 1
+#endif
+#ifndef GLOB_ONLYDIR
+#define GLOB_ONLYDIR 0
+#endif
+#endif
+// CYGWIN ADDED LINES END 
 
 #ifndef MONIT_H
 #define MONIT_H
